StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/ViewControllers/UpdatePaymentMethodViewController.swift

```diff
@@ -0,0 +1,378 @@
+            configuration.contentInsets = NSDirectionalEdgeInsets(top: 10, leading: 16, bottom: 10, trailing: 16)
+            configuration.baseBackgroundColor = .clear
+            button.configuration = configuration
+        } else {
+            button.contentEdgeInsets = UIEdgeInsets(top: 10, left: 16, bottom: 10, right: 16)
+        }
+        button.setTitleColor(appearance.colors.danger, for: .normal)
+        button.layer.borderColor = appearance.colors.danger.cgColor
+        button.layer.borderWidth = appearance.primaryButton.borderWidth
+        button.layer.cornerRadius = appearance.cornerRadius
+        button.setTitle(.Localized.remove, for: .normal)
+        button.titleLabel?.textAlignment = .center
+        button.titleLabel?.font = appearance.scaledFont(for: appearance.font.base.medium, style: .callout, maximumPointSize: 25)
+        button.titleLabel?.adjustsFontForContentSizeCategory = true
+        button.addTarget(self, action: #selector(removePaymentMethod), for: .touchUpInside)
+        button.isHidden = !viewModel.canRemove
+        return button
+    }()
+
+    private lazy var paymentMethodDetails: UIStackView = {
+        lazy var details: UIView = {
+            switch viewModel.paymentMethod.type {
+            case .card:
+                return cardSection.view
+            case .USBankAccount:
+                return usBankAccountSection
+            case .SEPADebit:
+                return sepaDebitSection
+            default:
+                fatalError("Updating payment method has not been implemented for \(viewModel.paymentMethod.type)")
+            }
+        }()
+        let paymentMethodDetails = UIStackView(arrangedSubviews: [details, detailsCannotBeChangedLabel])
+        paymentMethodDetails.axis = .vertical
+        paymentMethodDetails.setCustomSpacing(8, after: details) // custom spacing from figma
+        return paymentMethodDetails
+    }()
+
+    private lazy var detailsCannotBeChangedLabel: UITextView = {
+        let label = ElementsUI.makeSmallFootnote(theme: appearance.asElementsTheme)
+        label.text = viewModel.detailsCannotBeChanged
+        if viewModel.canEdit {
+            label.isHidden = true
+        }
+        else {
+            // if expiry date shows an error, don't show the label
+            if viewModel.paymentMethod.type == .card {
+                switch cardSection.validationState {
+                case .valid:
+                    label.isHidden = false
+                default:
+                    label.isHidden = true
+                }
+            }
+            label.isHidden = false
+        }
+        return label
+    }()
+
+    private lazy var errorLabel: UILabel = {
+        let label = ElementsUI.makeErrorLabel(theme: appearance.asElementsTheme)
+        label.isHidden = true
+        return label
+    }()
+
+    private lazy var cardBrandDropDown: DropdownFieldElement? = {
+        if viewModel.paymentMethod.isCoBrandedCard {
```

Comment:
This would be easier to read if we did

```swift
guard viewModel.paymentMethod.isCoBrandedCard else { return nil }
/// continue creating dropdown.
```